tests/unit/cdomains/threadIdDomainTest.ml

```diff
@@ -0,0 +1,106 @@
+
+  (* unique creates non-unique and prefixes are incompatible *)
+  assert_equal false (may_be_ancestor (main >> a) (main >> b >> a >> a)); (* 53-races-mhp/05-not-created3 *)
+  assert_equal false (may_be_ancestor (main >> a >> b) (main >> b >> a >> c >> c)); (* infeasible for race: definitely_not_started requires (main >> a), where this must happen, to be must parent for (main >> b >> a >> c >> c), which it is not *)
+  assert_equal false (may_be_ancestor (main >> a >> b) (main >> a >> c >> d >> d)); (* 53-races-mhp/06-not-created4, also passes with simple may_be_ancestor *)
+
+  (* non-unique creates non-unique: prefix must not lengthen *)
+  assert_equal false (may_be_ancestor (main >> a >> a) (main >> a >> b >> b)); (* infeasible for race: cannot create non-unique (main >> a >> a) before unique prefix-ed (main >> a >> b >> b) *)
+  assert_equal false (may_be_ancestor (main >> a >> a) (main >> b >> a >> a)); (* 53-races-mhp/07-not-created5 *)
+  (* non-unique creates non-unique: prefix must be compatible *)
+  assert_equal false (may_be_ancestor (main >> a >> b >> c >> c) (main >> b >> a >> c >> c)); (* infeasible for race: definitely_not_started requires (main >> a >> b or main >> a >> b >> c), where this must happen, to be must parent for (main >> b >> a >> c >> c), which it is not *)
+  (* non-unique creates non-unique: elements must not be removed *)
+  assert_equal false (may_be_ancestor (main >> a >> b >> b) (main >> a >> c >> c)); (* from set *) (* 53-races-mhp/08-not-created6, also passes with simple may_be_ancestor *)
+  assert_equal false (may_be_ancestor (main >> a >> b >> b) (main >> b >> b)); (* from prefix *) (* infeasible for race: definitely_not_started requires (main >> a or main >> a >> b), where this must happen, to be must parent for (main >> b >> b), which it is not *)
+  (* non-unique creates non-unique: removed elements and set must be in new set *)
+  (* assert_equal false (may_be_ancestor (main >> a >> b >> c >> c) (main >> a >> c >> c)); *)
+  (* TODO: cannot test due because by construction after prefix check? *)
+  (* non-unique creates non-unique *)
+  assert_equal true (may_be_ancestor (main >> a >> a) (main >> a >> a));
+  assert_equal true (may_be_ancestor (main >> a >> a) (main >> a >> a >> b));
+  assert_equal true (may_be_ancestor (main >> a >> a) (main >> a >> b >> a));
+  assert_equal true (may_be_ancestor (main >> a >> a) (main >> a >> b >> c >> a));
+  assert_equal true (may_be_ancestor (main >> a >> b >> b) (main >> a >> b >> b));
+  assert_equal true (may_be_ancestor (main >> a >> b >> b) (main >> a >> a >> b));
+  assert_equal true (may_be_ancestor (main >> a >> b >> b) (main >> a >> b >> a));
+  assert_equal true (may_be_ancestor (main >> a >> b >> b) (main >> b >> b >> a));
+  assert_equal true (may_be_ancestor (main >> a >> b >> b) (main >> b >> a >> b));
+
+  (* 4f6a7637b8d0dc723fe382f94bed6c822cd4a2ce passes all... *)
```

Comment:
Why this comment?

Reply:
I think the reason was that there are at least two improvements after that commit (dbd487438a573a845c449e7a5619a70cccdb6cf9 and c2c596f43798f081cbeadd6997c7a9b07da43bba) which are supposed to improve precision, but they're not observed by the tests. That might possibly be related to the other unit test TODOs about certain cases being untestable by construction.
Since then I refactored the code, so perhaps it's no longer true. At least according to code coverage, all cases seem to be covered by something.

Reply:
Then we can maybe just remove the comment?